### PR TITLE
docs: sd-radio-group docs [skip chromatic]

### PR DESCRIPTION
**Pull request by smfonseca**
## Definition of Reviewable:

- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

### CODE REVIEW

**Comment by mariohamann** [comment]
Would be great to have `setCustomValidation` in the docs instead of the screenshot tests.

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 1]
```diff
@@ -41,280 +48,97 @@ export default {
...
-        y: { type: 'attribute', name: 'size' }
-      },
-      args
-    });
-  }
-};
-
-/**
- * Use the `invalid` attribute to mark the element is not in a format or a value the application will accept.
+ * Use the `size` attribute to set the size of the radio buttons. The default is `lg`. In tight spaces, consider using `sm`.
```
Comment (coraliefeil) [inline packages/components/src/components/radio-group/radio-group.stories.ts:58]: Use the `size` attribute to change the size.

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 2]
```diff
@@ -41,280 +48,97 @@ export default {
...
-  play: async ({ canvasElement }: { canvasElement: HTMLUnknownElement }) => {
-    const el = canvasElement.querySelector('sd-button');
-    await waitUntil(() => el?.shadowRoot?.querySelector('button'));
-    await userEvent.type(el!.shadowRoot!.querySelector('button')!, '{return}', { pointerEventsCheck: 0 });
-  }
+export const Size = {
+  name: 'Size',
+  render: () => html`
+    <div class="flex gap-12">
+      <sd-radio-group name="radio-group" size="lg">
```
Comment (coraliefeil) [inline packages/components/src/components/radio-group/radio-group.stories.ts:65]: Hide label

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 3]
```diff
@@ -41,280 +48,97 @@ export default {
...
+        <sd-radio value="1">Radio 1</sd-radio>
+        <sd-radio value="2">Radio 2</sd-radio>
+        <sd-radio value="3">Radio 3</sd-radio>
+        <label slot="label">Large Group</label>
```
Comment (coraliefeil) [inline packages/components/src/components/radio-group/radio-group.stories.ts:69]: Hide label

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 4]
```diff
@@ -41,280 +48,97 @@ export default {
...
+      </sd-radio-group>
+
+      <sd-radio-group name="radio-group" size="sm">
+        <sd-radio value="1">Radio 1</sd-radio>
+        <sd-radio value="2">Radio 2</sd-radio>
+        <sd-radio value="3">Radio 3</sd-radio>
+        <label slot="label">Small Group</label>
```
Comment (coraliefeil) [inline packages/components/src/components/radio-group/radio-group.stories.ts:76]: Hide label

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 5]
```diff
@@ -41,280 +48,97 @@ export default {
...
-    });
-  }
+  name: 'Required',
+  render: () => html`
+    <div class="flex gap-12">
+      <sd-radio-group name="radio-group" required>
+        <sd-radio value="1">Radio 1</sd-radio>
+        <sd-radio value="2">Radio 2</sd-radio>
+        <sd-radio value="3">Radio 3</sd-radio>
+        <label slot="label">Required Group</label>
```
Comment (coraliefeil) [inline packages/components/src/components/radio-group/radio-group.stories.ts:97]: Pls bold group label

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 6]
```diff
@@ -41,280 +48,97 @@ export default {
...
+
+      <sd-radio-group name="radio-group" size="sm">
+        <sd-radio value="1">Radio 1</sd-radio>
+        <sd-radio value="2">Radio 2</sd-radio>
+        <sd-radio value="3">Radio 3</sd-radio>
+        <label slot="label">Small Group</label>
+      </sd-radio-group>
+    </div>
+  `
 };
```
Comment (coraliefeil) [inline packages/components/src/components/radio-group/radio-group.stories.ts:81]: I would move this "required" section underneath "orientation" so it comes last.

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 7]
```diff
@@ -41,280 +48,97 @@ export default {
...
 
 /**
- * 1. You can use the `setCustomValidity` method to set a custom validation message. This will override any native validation messages.
- * 2. Set an empty string to clear the custom validity and make the input valid.
- * 3. To show the validation message, call the `reportValidity` method. Originally this would show a native validation bubble, but we show the error messages inline.
+ * Use the `boldlabel` attribute to set the group label to bold.
```
Comment (coraliefeil) [inline packages/components/src/components/radio-group/radio-group.stories.ts:129]: I’m confused about this attribute ... why do we have that? Group label is always bold ... there should not be an attribute for that.

Comment (coraliefeil) [reply packages/components/src/components/radio-group/radio-group.stories.ts:129]: I would like to remove the attribute.

**Comment by coraliefeil** [comment]
Could we add Invalid like we did for checkbox?
Bildschirmfoto 2024-08-10 um 21 51 56

File: packages/components/src/components/radio-group/radio-group.stories.ts (hunk 1 above)
Comment (smfonseca) [reply packages/components/src/components/radio-group/radio-group.stories.ts]: Updated

File: packages/components/src/components/radio-group/radio-group.stories.ts (hunk 4 above)
Comment (smfonseca) [reply packages/components/src/components/radio-group/radio-group.stories.ts]: Updated

File: packages/components/src/components/radio-group/radio-group.stories.ts (hunk 5 above)
Comment (smfonseca) [reply packages/components/src/components/radio-group/radio-group.stories.ts]: Updated

File: packages/components/src/components/radio-group/radio-group.stories.ts (hunk 6 above)
Comment (smfonseca) [reply packages/components/src/components/radio-group/radio-group.stories.ts:79]: Updated

File: packages/components/src/components/radio-group/radio-group.stories.ts (hunk 7 above)
Comment (smfonseca) [reply packages/components/src/components/radio-group/radio-group.stories.ts:127]: Hidden as agreed in our daily sync and created a [ticket](https://github.com/solid-design-system/solid/issues/1274) to address it

**Comment by coraliefeil** [comment]
> > Could we add Invalid like we did for checkbox? Bildschirmfoto 2024-08-10 um 21 51 56
> 
> We already have `invalid` in the sd-radio docs. I don't think it's worth adding it here since these docs are for the group. What do you think?

Since radios always come in groups and mandatory error-text for invalid state needs to be shown underneath the radio group ... showing invalid is important.

File: packages/components/src/components/radio-group/radio-group.stories.ts (hunk 7 above)
Comment (mariohamann) [reply packages/components/src/components/radio-group/radio-group.stories.ts]: +1
I was shocked, that it exists. 😂

**Comment by mariohamann** [comment]
> > Could we add Invalid like we did for checkbox? Bildschirmfoto 2024-08-10 um 21 51 56
> 
> We already have `invalid` in the sd-radio docs. I don't think it's worth adding it here since these docs are for the group. What do you think?

I would only show it in the group, what do you think?

**Review by yoezlem — CHANGES_REQUESTED** [review]
A small thing - could you please set the initial check on Radio 2 to Radio 1 as shown in Figma?

Storybook:
image

Figma: 
image

**Review by yoezlem — CHANGES_REQUESTED** [review]
Whats with the "Releated Templates" Section? Its missing
image

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 8]
```diff
@@ -41,280 +48,79 @@ export default {
...
+        <sd-radio value="3">Radio 3</sd-radio>
+      </sd-radio-group>
+
+      <sd-radio-group name="radio-group" size="sm">
+        <sd-radio value="1">Radio 1</sd-radio>
+        <sd-radio value="2">Radio 2</sd-radio>
+        <sd-radio value="3">Radio 3</sd-radio>
+      </sd-radio-group>
+    </div>
+  `
```
Comment (yoezlem) [inline packages/components/src/components/radio-group/radio-group.stories.ts:77]: Please "check" the first <sd-radio>.
image

**Review by yoezlem — COMMENTED** [review]
Where is the label Story? Its not marked as Figma-only

image

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 9]
```diff
@@ -41,280 +48,79 @@ export default {
...
+      </sd-radio-group>
+
+      <sd-radio-group name="radio-group" orientation="horizontal">
+        <sd-radio value="1">Radio 1</sd-radio>
+        <sd-radio value="2">Radio 2</sd-radio>
+        <sd-radio value="3">Radio 3</sd-radio>
+        <label slot="label">Horizontal Group</label>
+      </sd-radio-group>
+    </div>
+  `
```
Comment (yoezlem) [inline packages/components/src/components/radio-group/radio-group.stories.ts:102]: https://github.com/solid-design-system/solid/pull/1257#discussion_r1721662496

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 10]
```diff
@@ -41,280 +48,79 @@ export default {
...
+  render: () => html`
+    <div class="flex gap-12">
+      <sd-radio-group name="radio-group" required boldlabel>
+        <sd-radio value="1">Radio 1</sd-radio>
+        <sd-radio value="2">Radio 2</sd-radio>
+        <sd-radio value="3">Radio 3</sd-radio>
+        <label slot="label">Required Group</label>
+      </sd-radio-group>
+    </div>
+  `
```
Comment (yoezlem) [inline packages/components/src/components/radio-group/radio-group.stories.ts:123]: https://github.com/solid-design-system/solid/pull/1257#discussion_r1721662496

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 11]
```diff
@@ -41,280 +48,79 @@ export default {
...
 };
 
 /**
  * Use the `required` attribute to mark the element as required. This can be used for form validation purposes.
+ *
+ *
+ * **Accessibility:** To ensure screen-reader compatibility, consider including a statement such as "Fields marked with an asterisk (*) are required" at the start of the form.
```
Comment (yoezlem) [inline packages/components/src/components/radio-group/radio-group.stories.ts:109]: I guess this is only Code relevant?
@coraliefeil @MartaPintoTeixeira will you add this to figma and mark it as code-only (or dev/storybook - don't know the exact wording)?

**Review by yoezlem — CHANGES_REQUESTED** [review]
I am also missing the stories: 
Disabled and Invalid.
image

**Comment by Vahid1919** [comment]
> Whats with the "Releated Templates" Section? Its missing image

These templates don't exist on code side, so I imagine that is why Sergio did not include them/

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 12]
```diff
@@ -41,217 +48,130 @@ export default {
...
+        <sd-radio value="2">Radio 2</sd-radio>
+        <sd-radio value="3">Radio 3</sd-radio>
+        <label slot="label">Required Group</label>
+      </sd-radio-group>
+
+      <div class="flex gap-2">
+        <sd-button class="w-min" type="submit">Submit</sd-button>
+        <sd-button class="w-min" type="reset" variant="secondary">Reset</sd-button>
+      </div>
+    </form>
```
Comment (yoezlem) [inline packages/components/src/components/radio-group/radio-group.stories.ts:142]: Can you add more space between radio group and the button?
image

**Review by yoezlem — CHANGES_REQUESTED** [review]
Can you change the order of size and label? First size than label. thx.

image

**Review by yoezlem — CHANGES_REQUESTED** [review]
We have a mismatch between Design and Code.

In Figma we have Invalid:
image

In Storybook Custom Validation:
image

Do we not need the validation story from Figma in Story? If not, than we have to mark it as Figma only.
And I guess "Custom Validation" is Storybook only. @coraliefeil @MartaPintoTeixeira should we add this to Figma? Otherwise, we risk having completely different content again.

**Comment by Vahid1919** [comment]
> We have a mismatch between Design and Code.
> 
> In Figma we have Invalid: image
> 
> In Storybook Custom Validation: image
> 
> Do we not need the validation story from Figma in Story? If not, than we have to mark it as Figma only. And I guess "Custom Validation" is Storybook only. @coraliefeil @MartaPintoTeixeira should we add this to Figma? Otherwise, we risk having completely different content again.

Adapting invalid story directly did not make sense development-side because we need the user to interact with/submit the form in order to show the invalid state. That's why @mariohamann  and I discussed reworking the Required story and adding the Custom Validation we have now :)

**Comment by coraliefeil** [comment]
I deleted the related templates for radio-group and radio-button-group in figma. Uneccessary effort.

File: packages/components/src/components/radio-group/radio-group.stories.ts (hunk 11 above)
Comment (coraliefeil) [reply packages/components/src/components/radio-group/radio-group.stories.ts:124]: We have this info in the accessibility section here in [Figma](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=2213-20986&t=AcHb8BRC0LXsi0AC-11).

We just gathered all a11y info for each comp in this section. We need to rework that at a later point ...

**Comment by coraliefeil** [comment]
Label
Use the label attribute to add a label on top. You can use the optional boldLabel attribute to make the label bold.
-> I already mentioned that the label has to be bold for better emphasize. This is not an option.

Please adjust the sentence to:

Use the label attribute to add a label on top.

**Comment by coraliefeil** [comment]
I don’t know what to do with the invalid section.

We agreed in our worksession to:

Invalid
The component gets “invalid” state when the form is not valid.

Should we move custom validation to "General" section? Otherwise we would repeat ourselves for all comps that have an invalid state.

**Comment by Vahid1919** [comment]
> I don’t know what to do with the invalid section.
> 
> We agreed in our worksession to:
> 
> Invalid The component gets “invalid” state when the form is not valid.
> 
> Should we move custom validation to "General" section? Otherwise we would repeat ourselves for all comps that have an invalid state.

I think Custom Validation differentiates itself since the emphasis is being able to set a custom validation text. 

> Label Use the label attribute to add a label on top. You can use the optional boldLabel attribute to make the label bold. -> I already mentioned that the label has to be bold for better emphasize. This is not an option.
> 
> Please adjust the sentence to:
> 
> Use the label attribute to add a label on top.

@coraliefeil the issue here is that the boldLabel attribute is required from dev side to make the label bold, so perhaps we should include that in the docs, OR we could deprecate it and make it bold by default. I feel like not mentioning it at all, might create an inconsistency between the docs and component. 
 
 What do you think @mariohamann ? One more thing, if we do decide to deprecate it, should it be part of this PR or a new issue?

**Comment by mariohamann** [comment]
In future we should just make `<b>Label</b>` to make something bold... not sure, how this slipped in. I don't think it's too bad, to mention it in the label area at the moment. Deprecation and removement should happen in future, let's mark it as Breaking Change, so we can collect things.

**Comment by mariohamann** [comment]
> Should we move custom validation to "General" section? Otherwise we would repeat ourselves for all comps that have an invalid state.

You're absolutely right. We need a section like "Form handling". Let's copy paste for the moemnt, and we have the time to buld the other page up, this would be great – but at least then we have it somewhere and improve in future. I think it's fine, to only document that on Dev side.

(Shoelace does it the same with a explicit section for all components: https://shoelace.style/getting-started/form-controls)

**Comment by Vahid1919** [comment]
> In future we should just make `<b>Label</b>` to make something bold... not sure, how this slipped in. I don't think it's too bad, to mention it in the label area at the moment. Deprecation and removement should happen in future, let's mark it as Breaking Change, so we can collect things.

What do you think @coraliefeil? Since the boldLabel would be a breaking change, for now we can have something like: 

Use the label attribute to add a label on top. Additionally, and the boldLabel attribute is required to ensure the label stands out for better readability and UI consistency.


Then later on we can deprecate the boldLabel attribute and update the docs to reflect that.

**Comment by coraliefeil** [comment]
> > In future we should just make `<b>Label</b>` to make something bold... not sure, how this slipped in. I don't think it's too bad, to mention it in the label area at the moment. Deprecation and removement should happen in future, let's mark it as Breaking Change, so we can collect things.
> 
> What do you think @coraliefeil? Since the boldLabel would be a breaking change, for now we can have something like:
> 
> Use the label attribute to add a label on top. Additionally, and the boldLabel attribute is required to ensure the label stands out for better readability and UI consistency.
> 
> Then later on we can deprecate the boldLabel attribute and update the docs to reflect that.

Ok in code. In figma we’ll just keep "Use the label attribute to add a label on top." No breaking change on our end.

**Review by mariohamann — APPROVED** [review]

**Review by mariohamann — APPROVED** [review]

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 13]
```diff
@@ -41,280 +48,156 @@ export default {
...
-  }
+export const Orientation = {
+  name: 'Orientation',
+  render: () => html`
+    <div class="flex gap-12">
+      <sd-radio-group name="radio-group" orientation="vertical" value="1">
+        <sd-radio value="1">Radio 1</sd-radio>
+        <sd-radio value="2">Radio 2</sd-radio>
+        <sd-radio value="3">Radio 3</sd-radio>
+        <label slot="label">Vertical Group</label>
```
Comment (coraliefeil) [inline packages/components/src/components/radio-group/radio-group.stories.ts:107]: Don’t show group label

File: packages/components/src/components/radio-group/radio-group.stories.ts [hunk 14]
```diff
@@ -41,280 +48,156 @@ export default {
...
+      </sd-radio-group>
+
+      <sd-radio-group name="radio-group" orientation="horizontal" value="1">
+        <sd-radio value="1">Radio 1</sd-radio>
+        <sd-radio value="2">Radio 2</sd-radio>
+        <sd-radio value="3">Radio 3</sd-radio>
+        <label slot="label">Horizontal Group</label>
```
Comment (coraliefeil) [inline packages/components/src/components/radio-group/radio-group.stories.ts:114]: Don’t show group label

**Comment by coraliefeil** [comment]
Pls adress my last comments and then it’s approved from design side.

**Comment by yoezlem** [comment]
> > Should we move custom validation to "General" section? Otherwise we would repeat ourselves for all comps that have an invalid state.
> 
> You're absolutely right. We need a section like "Form handling". Let's copy paste for the moemnt, and we have the time to buld the other page up, this would be great – but at least then we have it somewhere and improve in future. I think it's fine, to only document that on Dev side.
> 
> (Shoelace does it the same with a explicit section for all components: https://shoelace.style/getting-started/form-controls)

Please create a ticket for that.

**Comment by coraliefeil** [comment]
> > > Should we move custom validation to "General" section? Otherwise we would repeat ourselves for all comps that have an invalid state.
> > 
> > 
> > You're absolutely right. We need a section like "Form handling". Let's copy paste for the moemnt, and we have the time to buld the other page up, this would be great – but at least then we have it somewhere and improve in future. I think it's fine, to only document that on Dev side.
> > (Shoelace does it the same with a explicit section for all components: https://shoelace.style/getting-started/form-controls)
> 
> Please create a ticket for that.

Done

https://github.com/solid-design-system/solid/issues/1313

**Pull request merged**